lib/client/api.go

```diff
@@ -3241,39 +3239,23 @@ func (tc *TeleportClient) SSHLogin(ctx context.Context, sshLoginFunc SSHLoginFun
 }
 
 // GetNewLoginKey gets a new private key for login.
-func (tc *TeleportClient) GetNewLoginKey(ctx context.Context, keyPolicy keys.PrivateKeyPolicy) (*keys.PrivateKey, error) {
-	key, err := tc.LocalAgent().GetCoreKey()
-	if err == nil {
-		// If we find an existing key with a non-zero key polic and it meets
-		// the given keyPolicy requirement, then we should use the existing key.
-		if coreKeyPolicy := keys.GetPrivateKeyPolicy(key.PrivateKey); coreKeyPolicy != keys.PrivateKeyPolicyNone {
-			if err := keyPolicy.VerifyPolicy(coreKeyPolicy); err == nil {
-				return key.PrivateKey, nil
-			}
-		}
```

Comment:
why remove the check for an existing key that satisfies the policy?

Reply:
With this change, the `~/.tsh/proxy.yaml` file should always have an accurate `PrivateKeyPolicy` field. Its more efficient for us to check the yaml file than load up the private key, and connect with PIV, just to check the policy.

Let me know if you think I missed an edge case, but I couldn't think of any, outside of the user deleting/modifying `~/.tsh/proxy.yaml` after the user logs in.